lib/smart_todo/events.rb

```diff
@@ -64,15 +76,41 @@ def issue_close(organization, repo, issue_number)
     # @param pr_number [String, Integer]
     # @return [false, String]
     def pull_request_close(organization, repo, pr_number)
-      IssueClose.new(organization, repo, pr_number, type: "pulls").met?
+      IssueClose.new(organization, repo, pr_number, github_client, type: "pulls").met?
     end
 
     # Check if the installed ruby version meets requirements.
     #
     # @param requirements [Array<String>] a list of version specifiers
     # @return [false, String]
     def ruby_version(*requirements)
-      RubyVersion.new(requirements).met?
+      RubyVersion.new(requirements, installed_ruby_version).met?
+    end
+
+    private
+
+    def now
+      @now ||= Time.now
+    end
+
+    def spec_set
+      @spec_set ||= Bundler.load.specs
+    end
+
+    def rubygems_client
+      @rubygems_client ||= Net::HTTP.new("rubygems.org", Net::HTTP.https_default_port).tap do |client|
+        client.use_ssl = true
+      end
+    end
+
+    def github_client
+      @github_client ||= Net::HTTP.new("api.github.com", Net::HTTP.https_default_port).tap do |client|
+        client.use_ssl = true
+      end
+    end
+
+    def installed_ruby_version
+      @installed_ruby_version ||= Gem::Version.new(RUBY_VERSION)
```

Comment:
Why aren't we setting those in the initialize?

Reply:
We're lazily setting these up because if you don't find any todos that have a gem version, there's no need to instantiate these objects. Basically it mirrors the previous behavior of lazily creating objects based on the todos you find.